activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -453,6 +453,9 @@ def checkout_and_verify(c)
           c.verify!
         end
         c
+      rescue
+        disconnect!
```

Comment:
> This patch fix it by catching exception and releasing connection.

Aren't you forcefully disconnecting and releasing all of the pool's connections (instead of releasing just the "failed" connection)?